app/services/summary_service.rb

```diff
@@ -77,6 +77,22 @@ def self.new_arrival?(offender)
     end
   end
 
+  def self.approaching_handover?(offender)
+    today = Time.zone.today
+    thirty_days_time = today + 30.days
+
+    start_date = offender.handover_start_date
+    handover_date = offender.responsibility_handover_date
+
+    return false if start_date.nil?
+
+    if start_date.future?
+      start_date.between?(today, thirty_days_time)
+    else
+      today.between?(start_date, handover_date)
+    end
+  end
+
```

Comment:
I think this method is re-expressing some of the logic found in the [handover date service](https://github.com/ministryofjustice/offender-management-allocation-manager/blob/master/app/services/handover_date_service.rb). 

This method is probably clearer than our existing implementation! But I'm worried that we'll fail to change both in tandem, in the future, especially given the test coverage here.

Is there any good way to bring them together without extensive refactoring of the `handover_date_service`? I'm not sure, but bringing it up in case we have any good ideas! :sweat_smile:

Reply:
That could be the case, although my understanding was that the handover date service had a slightly different responsibility – being more concerned with calculating when the handover should happen, i.e. to calculate the handover dates themselves. (Maybe it should be the 'responsibility date calculator service'? 😅)

The "is this offender approaching handover?" question currently comes with the baked-in assumption that "approaching handover" means "handovers starting within the next 30 days". I'm not sure that "in the next 30 days" is a global definition of when a handover is approaching, or more specific to this 'handover cases' SPO summary page.

For example, POMs have a [disturbingly similar page](https://github.com/ministryofjustice/offender-management-allocation-manager/blob/master/app/controllers/caseload_controller.rb#L31-L35), to see offenders which are approaching handover. However it [doesn't follow exactly the same criteria](https://github.com/ministryofjustice/offender-management-allocation-manager/blob/master/app/controllers/caseload_controller.rb#L39-L53): it doesn't include offenders who are already in their handover period.

I definitely agree that this could be cleaned up – especially given the similarities between POM/COM functionality. I'm not sure I have any good ideas for moving towards that goal in this PR 😅 but open to suggestions.

Reply:
I think the `handover_date_service` deals with both concepts - both "approaching" handover and the actual date responsibility switches from POM to COM. The [data it returns](https://github.com/ministryofjustice/offender-management-allocation-manager/blob/master/app/services/handover_date_service.rb#L4) refers to these as "start date" and "handover date" respectively.

The 30-day thing though, I think is new, and specific to this page. Argh. I think you're right that moving towards it in this PR isn't good, but I do think that those two core classes - `handover_date_service` and `responsibility_service` - need some love soon.